pkg/helm/actions/upgrade_release_test.go

```diff
@@ -0,0 +1,95 @@
+package actions
+
+import (
+	"io/ioutil"
+	"testing"
+
+	"helm.sh/helm/v3/pkg/action"
+	"helm.sh/helm/v3/pkg/chart"
+	"helm.sh/helm/v3/pkg/chartutil"
+	kubefake "helm.sh/helm/v3/pkg/kube/fake"
+	"helm.sh/helm/v3/pkg/release"
+	"helm.sh/helm/v3/pkg/storage"
+	"helm.sh/helm/v3/pkg/storage/driver"
+	"helm.sh/helm/v3/pkg/time"
+)
+
+func TestUpgradeRelease(t *testing.T) {
+	tests := []struct {
+		releaseName  string
+		chartPath    string
+		chartName    string
+		chartVersion string
+	}{
+		{
+			releaseName:  "valid chart path",
+			chartPath:    "../testdata/influxdb-3.0.2.tgz",
+			chartName:    "influxdb",
+			chartVersion: "3.0.2",
+		},
+		{
+			releaseName:  "invalid chart path",
+			chartPath:    "../testdata/influxdb-3.0.1.tgz",
+			chartName:    "influxdb",
+			chartVersion: "3.0.2",
+		},
+	}
+	for _, tt := range tests {
+		t.Run(tt.releaseName, func(t *testing.T) {
+			store := storage.Init(driver.NewMemory())
+			actionConfig := &action.Configuration{
+				Releases:     store,
+				KubeClient:   &kubefake.PrintingKubeClient{Out: ioutil.Discard},
+				Capabilities: chartutil.DefaultCapabilities,
+				Log:          func(format string, v ...interface{}) {},
+			}
+
+			r := release.Release{
+				Name:      "test",
+				Namespace: "test-namespace",
+				Info: &release.Info{
+					FirstDeployed: time.Time{},
+					Status:        "deployed",
+				},
+				Version: 1,
+				Chart: &chart.Chart{
+					Metadata: &chart.Metadata{
+						Name:    "influxdb",
+						Version: "3.0.2",
+					},
+				},
+			}
+
+			store.Create(&r)
+
+			rel, err := UpgradeRelease("test-namespace", "test", tt.chartPath, nil, actionConfig)
+			if tt.releaseName == "valid chart path" {
```

Comment:
same above.. please align the structure of the table tests with the existing one

Reply:
Done